tracing-subscriber/src/layer/mod.rs

```diff
@@ -0,0 +1,1183 @@
+//! event will be recorded if it is enabled by _any_ per-layer filter, but it
+//! will be skipped by the layers whose filters did not enable it. Building on
+//! the previous example:
+//!
+//! ```
+//! use tracing_subscriber::{filter::{filter_fn, LevelFilter}, prelude::*};
+//!
+//! let access_log = // ...
+//!     # LevelFilter::INFO;
+//! let fmt_layer = tracing_subscriber::fmt::layer();
+//!
+//! tracing_subscriber::registry()
+//!     // Add the filter for the "http_access" target to the access
+//!     // log layer, like before.
+//!     .with(access_log.with_filter(filter_fn(|metadata| {
+//!         metadata.target() == "http_access"
+//!     })))
+//!     // Add a filter for spans and events with the INFO level
+//!     // and below to the logging layer.
+//!     .with(fmt_layer.with_filter(LevelFilter::INFO))
+//!     .init();
+//!
+//! // Neither layer will observe this event
+//! tracing::debug!(does_anyone_care = false, "a tree fell in the forest");
+//!
+//! // This event will be observed by the logging layer, but not
+//! // by the access log layer.
+//! tracing::warn!(dose_roentgen = %3.8, "not great, but not terrible");
+//!
+//! // This event will be observed only by the access log layer.
+//! tracing::trace!(target: "http_access", "HTTP request started");
+//!
+//! // Both layers will observe this event.
+//! tracing::error!(target: "http_access", "HTTP request failed with a very bad error!");
+//! ```
+//!
+//! A per-layer filter can be applied to multiple [`Layer`]s at a time, by
+//! combining them into a [`Layered`] layer using [`Layer::and_then`], and then
+//! calling [`Layer::with_filter`] on the resulting [`Layered`] layer.
+//!
+//! For example, if two layers `layer_a` and `layer_b` should only recieve the
+//! [`INFO`] [level] and above, but a third `layer_c` should recieve spans and
+//! events with the [`DEBUG`] [level]:
+//!
+//! ```
+//! use tracing_subscriber::{filter::LevelFilter, prelude::*};
+//!
+//! let layer_a = // ...
+//! # LevelFilter::INFO;
+//! let layer_b =  // ...
+//! # LevelFilter::INFO;
+//! let layer_c =  // ...
+//! # LevelFilter::INFO;
+//!
+//! let info_layers = layer_a
+//!     // Combine `layer_a` and `layer_b` into a `Layered` layer:
+//!     .and_then(layer_b)
+//!     // ...and then add an `INFO` `LevelFilter` to that layer:
+//!     .with_filter(LevelFilter::INFO);
+//!
+//! tracing_subscriber::registry()
+//!     // Add `layer_c` with a `DEBUG` filter.
+//!     .with(layer_c.with_filter(LevelFilter::DEBUG))
+//!     .with(info_layers)
+//!     .init();
+//!```
+//!
+//! If a [`Filtered`] [`Layer`] is combined with another [`Layer`]
+//! [`Layer::and_then`], and a filter is added to the [`Layered`] layer, that
+//! layer will be filtered by *both* the inner filter and the outer filter
+//! &mdash; only spans and events that are enabled by *both* filters will be
+//! observed by that layer. This can be used to implement complex filtering
+//! trees.
```

Comment:
```suggestion
//! If a [`Filtered`] [`Layer`] is combined with another [`Layer`]
//! [`Layer::and_then`], and a filter is added to the [`Layered`] layer, that
//! layer will be filtered by *both* the inner filter and the outer filter.
//! Only spans and events that are enabled by *both* filters will be
//! observed by that layer. This can be used to implement complex filtering
//! trees.
```